recipes/cgal/all/conanfile.py

```diff
@@ -14,63 +17,162 @@ class CgalConan(ConanFile):
+    def export_sources(self):
+        export_conandata_patches(self)
 
-    def _patch_sources(self):
-        if scm.Version(self.version) < "5.3":
-            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                "CMAKE_SOURCE_DIR", "CMAKE_CURRENT_SOURCE_DIR")
-        else:
-            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                "if(NOT PROJECT_NAME)", "if(TRUE)")
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.75.0")
-        self.requires("eigen/3.3.9")
+        self.requires("boost/1.78.0")
+        self.requires("eigen/3.4.0")
         self.requires("mpfr/4.1.0")
 
     def package_id(self):
         self.info.clear()
 
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._minimum_compilers_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.",
+            )
+
+    def _patch_sources(self):
+        replace_in_file(self,  os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "if(NOT PROJECT_NAME)", "if(1)", strict=False)
+        for it in self.conan_data.get("patches", {}).get(self.version, []):
+            patch(self, **it, strip=2)
+
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
 
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
-        files.rmdir(self, os.path.join(self.package_folder, "share"))
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
-        files.rmdir(self, os.path.join(self.package_folder, "bin"))
+        copy(self, "LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "bin"))
+        rm(self, "*Config*.cmake", os.path.join(self.package_folder, "lib", "cmake", "CGAL"))
+        rm(self, "Find*.cmake", os.path.join(self.package_folder, "lib", "cmake", "CGAL"))
```

Comment:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```

You could just remove the cmake folder instead.

Reply:
No I cannot. The directory `lib/cmake/CGAL` in the CGAL installation still contains a lot of files, and some of them (most of them) are very important for the CMake API of CGAL. For example, the `CMakeLists.txt` of the test package uses the interface CMake target `CGAL::Eigen3_support` provided in the file `CGAL_Eigen3_support.cmake`. Those are documented, see for example [the section **4.2 Eigen** in the documentation](https://doc.cgal.org/5.5.2/Manual/thirdparty.html#title8).

I have removed the `FindXXX.cmake` module, and `CGALConfig.cmake`, to respect the Conan policy.